bot/cogs/help_channels.py

```diff
@@ -546,13 +549,14 @@ async def move_to_dormant(self, channel: discord.TextChannel, caller: str) -> No
 
         self.bot.stats.incr(f"help.dormant_calls.{caller}")
 
-        if channel.id in self.claim_times:
-            claimed = self.claim_times[channel.id]
+        if await self.claim_times.contains(channel.id):
+            claimed_datestring = await self.claim_times.get(channel.id)
```

Comment:
Maybe should better to use just `get` and make `None` check instead first `contains` and then `get`.

Reply:
I agree; this is a potential race condition, as both the contains check and the get action are awaited. If another task "runs" after the contains was requested from Redis but before `get` is request from Redis, it could change the state we've just checked for.

Reply:
Yep, agreed.